create-custom-map-style/README.md

```diff
@@ -0,0 +1,990 @@
+
+```
+
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Sid": "PlaceIndexReadOnly",
+            "Effect": "Allow",
+            "Action": [
+                "geo:GetMapStyleDescriptor",
+                "geo:GetMapGlyphs",
+                "geo:GetMapSprites",
+                "geo:GetMapStyleDescriptor",
+                "geo:GetMapTile"
+            ],
+            "Resource": "arn:aws:geo:<YOUR ACCOUNT ID>:us-west-2:map/CreateCustomMapStyle"
+        },
+        {
+            "Sid": "MapsReadOnly",
+            "Effect": "Allow",
+            "Action": [
+                "geo:GetMapStyleDescriptor",
+                "geo:GetMapGlyphs",
+                "geo:GetMapSprites",
+                "geo:GetMapStyleDescriptor",
+                "geo:GetMapTile"
+            ],
+            "Resource": "arn:aws:geo:<YOUR ACCOUNT ID>:us-west-2:map/CreateCustomMapStyle"
+        }
+    ]
+}
+
+```
+
+### React - Amazon Location Service Integration
+
+1. Now it's time for adding yet another package; this time a very important library that uses WebGL to render interactive maps from [vector tiles](https://docs.mapbox.com/help/glossary/vector-tiles/) and [Mapbox styles](https://docs.mapbox.com/mapbox-gl-js/style-spec/)
+
+	```
+	npm install mapbox-gl --save
+	```
+
+2. With this we'll start adding the map to our page. Start by importing the dependencies in your `SamplePage` component
+
+	```
+	...
+	import  MapboxWorker  from  'worker-loader!mapbox-gl/dist/mapbox-gl-csp-worker';
+	import  mapboxgl  from  'mapbox-gl';
+	...
+	```
+
+3. Next fill out some boilerplate
+
+	```
+	...
+	// Below your import statements
+	mapboxgl.workerClass = MapboxWorker;
+	mapboxgl.accessToken = 'xx.xxX0XxxxXXxxxXXxXXXxXXXxXxxxX0x0XX01XXXxXX00xxXxx0X0x000x0X0XXX0.xX0X0XXX0x0XxXxXxx0xXx';
+	// This is the default Mapbox access token
+	...
+	```
+
+4.  Unfortunately Mapbox requires an access token to even instantiate their MapboxGL class, so be sure generate your own one [here](https://account.mapbox.com/access-tokens).
+
+5. Here's where we'll get a chance to fire up our react hooks. We need to add a reference (`useRef`) to the `.maputnik-map` DOM node, make an asynchronous API call (`useEffect`) to fetch data using Amplify, and keep track of the latitude, longitude, and zoom level of the map (`useState`)
+
+	```
+	...
+	import  React, { useRef, useState, useEffect } from  'react';
+	import { Auth } from  'aws-amplify';
+	...
+	const  SamplePage = () => {
+		const  maputnik = useRef();
+		const [maputnikCoordinates, setMaputnikCoordinates] = useState({
+			lat:  40.7356,
+			lng: -74.0541,
+			zoom:  12.44,
+		});
+		useEffect(async () => {
+			const  credentials = await  Auth.currentCredentials();
+			const { lat, lng, zoom } = maputnikCoordinates;
+			const  maputnikMap = new  mapboxgl.Map({
+				container:  maputnik.current,
+				center: { lng, lat },
+				zoom,
+				style:  'http://localhost:3000/example-style-descriptor.json',
```

Comment:
If this is published to S3/CF, this won't work.

Reply:
replaced this with `style: \`${process.env.PUBLIC_URL}/example-style-descriptor.json\`,`